android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt

```diff
@@ -443,86 +458,112 @@ class JsonConverter {
+                    json.putBoolean("isPictureInPictureAvailable", event.isPictureInPictureAvailable)
+                }
+
+                is PlayerEvent.AdBreakFinished -> {
+                    json.putMap("adBreak", fromAdBreak(event.adBreak))
+                }
+
+                is PlayerEvent.AdBreakStarted -> {
+                    json.putMap("adBreak", fromAdBreak(event.adBreak))
+                }
+
+                is PlayerEvent.AdClicked -> {
+                    json.putString("clickThroughUrl", event.clickThroughUrl)
+                }
+
+                is PlayerEvent.AdError -> {
+                    json.putInt("code", event.code)
+                    json.putString("message", event.message)
+                    json.putMap("adConfig", fromAdConfig(event.adConfig))
+                    json.putMap("adItem", fromAdItem(event.adItem))
+                }
+
+                is PlayerEvent.AdFinished -> {
+                    json.putMap("ad", fromAd(event.ad))
+                }
+
+                is PlayerEvent.AdManifestLoad -> {
+                    json.putMap("adBreak", fromAdBreak(event.adBreak))
+                    json.putMap("adConfig", fromAdConfig(event.adConfig))
+                }
+
+                is PlayerEvent.AdManifestLoaded -> {
+                    json.putMap("adBreak", fromAdBreak(event.adBreak))
+                    json.putMap("adConfig", fromAdConfig(event.adConfig))
+                    json.putDouble("downloadTime", event.downloadTime.toDouble())
+                }
+
+                is PlayerEvent.AdQuartile -> {
+                    json.putString("quartile", fromAdQuartile(event.quartile))
+                }
+
+                is PlayerEvent.AdScheduled -> {
+                    json.putInt("numberOfAds", event.numberOfAds)
+                }
+
+                is PlayerEvent.AdSkipped -> {
+                    json.putMap("ad", fromAd(event.ad))
+                }
+
+                is PlayerEvent.AdStarted -> {
+                    json.putMap("ad", fromAd(event.ad))
+                    json.putString("clickThroughUrl", event.clickThroughUrl)
+                    json.putString("clientType", fromAdSourceType(event.clientType))
+                    json.putDouble("duration", event.duration)
+                    json.putInt("indexInQueue", event.indexInQueue)
+                    json.putString("position", event.position)
+                    json.putDouble("skipOffset", event.skipOffset)
+                    json.putDouble("timeOffset", event.timeOffset)
+                }
+
+                is PlayerEvent.VideoPlaybackQualityChanged -> {
+                    json.putMap("newVideoQuality", fromVideoQuality(event.newVideoQuality))
+                    json.putMap("oldVideoQuality", fromVideoQuality(event.oldVideoQuality))
+                }
+
+                else -> {
+                    // Event is not supported yet or does not have any additional data
```

Comment:
A `Log` would be nice :)

Reply:
https://github.com/bitmovin/bitmovin-player-react-native/pull/240#discussion_r1329690343